source/common/router/router.cc

```diff
@@ -49,6 +49,11 @@ bool FilterUtility::shouldShadow(const ShadowPolicy& policy, Runtime::Loader& ru
     return false;
   }
 
+  if (policy.defaultValue().numerator() > 0) {
+    return runtime.snapshot().featureEnabled(policy.runtimeKey(), policy.defaultValue(),
+                                             stable_random);
+  }
+
   if (!policy.runtimeKey().empty() &&
       !runtime.snapshot().featureEnabled(policy.runtimeKey(), 0, stable_random, 10000UL)) {
     return false;
```

Comment:
Can we also set the `default_value_` denominator to `10,000` when we notice that the `runtime_fraction` isn't set?

Reply:
Sure, but still we can't merge these blocks because if the value for `runtime key` is not in fraction form (e.g.: it's an integer), the default denominator will still be 100:

https://github.com/envoyproxy/envoy/blob/master/source/common/runtime/runtime_impl.cc#L197

regardless of what the denominator for `default_value_` is. So that would break the expectation for current users of `runtime_key` — that's why I was thinking we should leave the if blocks separate.

Also, given that setting the denominator in `default_value_` to `10000` won't affect integers found in the runtime we could still do it, but for the case in which it's omitted in the config since it isn't required:

https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/percent.proto#L47

Reply:
Makes sense, let's keep them separated.